Enterprise-CMCS/eAPD
# 4540 [Feature] Trap errors on the react side to avoid the white screen of death

Resolves #4540 

### Description
When issues are thrown on the frontend, they currently cause a white screen of death and you have to do a hard refresh or remove parts of the URL to navigate back to a viewable page. If an error is thrown on the frontend the user is taken to an error page and they can navigate to a safe page from there.

### Chromatic Link
https://www.chromatic.com/build?appId=61d5b948cf6f17003a12bf77&number=1498

### Automated test cases written

| Given | When | Then | Type (jest, tap, cypress) |
| ----- | ---- | ---- | ------------------------- |
| An error is thrown | by incomplete code | should throw an error page where it's easier to go to another page | jest |

### Steps to manually verify this change

1. Go to a page, delete some code or put in bad code to case the page to not load
2. Confirm the below shows up instead of a blank screen.

<img width="1169" alt="Screenshot 2023-04-17 at 3 12 55 PM" src="https://user-images.githubusercontent.com/44708048/232587733-2c039c73-38bd-4ee3-b5ed-cfbde89a84e0.png">

### This pull request is ready to code review when

      tests, refactor
      screenreader, text scaling) OR an exemption is documented

### This pull request is ready to test when


### This pull request is ready to review when QA has


### This pull request can be merged when
